server/emails/templates/BaseEmail.tsx

```diff
@@ -101,11 +104,43 @@ export default abstract class BaseEmail<
       return;
     }
 
+    let references: string[] | undefined;
+
+    if (notification) {
+      const prevNotifications = await Notification.unscoped().findAll({
```

Comment:
Could we move all this logic to a new static method on `Notification`, something like `Notification.getReferencesForNotification(notification)`, I think this will keep things tidy and create a more testable interface

Reply:
Yes, much cleaner.